### PR TITLE
Combine simplified visitors

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Extracted from #57.

### CODE REVIEW

File: vendor-extra/ViewsBundle/src/Views/SimplifiedVisitor.php [hunk 1]
```diff
@@ -5,14 +5,20 @@
...
+use LogicException;
 use function in_array;
+use function is_string;
 use function sprintf;
 
 trait SimplifiedVisitor
 {
     final public function visit(Element $object, View $view) : View
     {
-        if ($this->expectedTemplate() !== $view->getTemplate()) {
```
Comment (giorgiosironi) [inline vendor-extra/ViewsBundle/src/Views/SimplifiedVisitor.php]: Tried to review from mobile and just didn't work.

Difference between `expectedTemplate` and `possibleTemplate` is unclear, can they be documented with docblocks here? There is no separate interface for them so this is the best place, also it's where all the dependencies fan in.

Comment (thewilkybarkid) [reply vendor-extra/ViewsBundle/src/Views/SimplifiedVisitor.php]: Might need to rethink this a little. Currently `expectedTemplate` is for cases where the `View` is requesting a specific template (eg ContentHeader), whereas `possibleTemplate` is for cases where the `View` might not know what template it needs (eg when cycling through body content, which contains paragraphs, sections, lists etc).

Comment (thewilkybarkid) [reply vendor-extra/ViewsBundle/src/Views/SimplifiedVisitor.php]: Had a go in 23fdf4a.

Comment (giorgiosironi) [reply vendor-extra/ViewsBundle/src/Views/SimplifiedVisitor.php]: This ~class~ trait now makes few assumptions about its children.

**Pull request merged**